catroid/src/main/java/org/catrobat/catroid/stage/VisualPlacementActivity.java

```diff
@@ -132,50 +135,66 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 		getSupportActionBar().setTitle(R.string.brick_place_at_option_place_visually);
 
 		if (ProjectManager.getInstance().isCurrentProjectLandscapeMode()) {
-			setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+			setRequestedOrientation(SCREEN_ORIENTATION_LANDSCAPE);
 		} else {
-			setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+			setRequestedOrientation(SCREEN_ORIENTATION_PORTRAIT);
 		}
 
 		FrameLayout frameLayout = findViewById(R.id.frame_container);
 		Project currentProject = ProjectManager.getInstance().getCurrentProject();
 		Sprite currentSprite = ProjectManager.getInstance().getCurrentSprite();
 
-		int currentScreenWidth = getResources().getDisplayMetrics().widthPixels;
-		int currentScreenHeight = getResources().getDisplayMetrics().heightPixels;
+		int screenWidth = ScreenValues.SCREEN_WIDTH;
+		int screenHeight = ScreenValues.SCREEN_HEIGHT;
 		int virtualScreenWidth = currentProject.getXmlHeader().virtualScreenWidth;
 		int virtualScreenHeight = currentProject.getXmlHeader().virtualScreenHeight;
 
-		final float scaleHeightRatio = (float) currentScreenHeight / (float) virtualScreenHeight;
-		reversedScaleHeightRatio = (float) virtualScreenHeight / (float) currentScreenHeight;
-		final float scaleWidthRatio = (float) currentScreenWidth / (float) virtualScreenWidth;
-		reversedScaleWidthRatio = (float) virtualScreenWidth / (float) currentScreenWidth;
 
+		float aspectRatio = (float) virtualScreenWidth / (float) virtualScreenHeight;
+		float screenAspectRatio = ScreenValues.getAspectRatio();
+
+		float scale;
+		float ratioHeight = (float) screenHeight / (float) virtualScreenHeight;
+		float ratioWidth = (float) screenWidth / (float) virtualScreenWidth;
 		ScreenModes projectScreenMood = currentProject.getXmlHeader().getScreenMode();
 		if (projectScreenMood == MAXIMIZE) {
```

Comment:
you could use a `switch (projectScreenMode) { case MAXIMIZE: ... } ` instead of the `if else`.

Reply:
as you wish.